module/localisation/RobotLocalisation/src/RobotLocalisation.cpp

```diff
@@ -349,15 +349,15 @@ namespace module::localisation {
             }
         }
 
-        return std::max(weight, 0.0);
+        return std::max(weight, 0.0f);
     }
 
     void RobotLocalisation::resample() {
-        std::vector<double> weights(particles.size());
+        std::vector<float> weights(particles.size());
         for (size_t i = 0; i < particles.size(); i++) {
             weights[i] = particles[i].weight;
         }
-        double weight_sum = std::accumulate(weights.begin(), weights.end(), 0.0);
+        float weight_sum = std::accumulate(weights.begin(), weights.end(), 0.0f);
         if (weight_sum == 0) {
```

Comment:
This is not the best comparison to make, the result of a floating point computation is rarely going to be exactly any specific number, you are better off checking for `std::abs(weight_sum) < epsilon`